code/modules/mob/living/simple_animal/bot/hygienebot.dm

```diff
@@ -0,0 +1,356 @@
+						speak(pick("Get back here you foul smelling fucker.", "If you don't get back here right now I'm going to give you a fucking vasectomy.", "STOP RUNNING OR I WILL CUT YOUR ARTERIES!", "Just fucking let me clean you you arsehole!", "STOP. RUNNING.", "Either you stop running or I will fucking drag you out of an airlock.", "I just want to fucking clean you you troglodyte.", "If you don't come back here I'll put a green cloud around you cunt."))
+					if((get_dist(src, target)) >= (olddist))
+						frustration++
+					else
+						frustration = 0
+			else
+				back_to_idle()
+
+		if(BOT_SHOWERSTANCE)
+			if(check_purity(target))
+				speak("Enjoy your clean and tidy day!")
+				playsound(loc, 'sound/effects/hygienebot_happy.ogg', 60, 1)
+				back_to_idle()
+				return
+			if(!target)
+				last_found = world.time
+			if(target.loc != loc && !isturf(target.loc))
+				back_to_hunt()
+
+		if(BOT_START_PATROL)
+			look_for_lowhygiene()
+			start_patrol()
+
+		if(BOT_PATROL)
+			look_for_lowhygiene()
+			bot_patrol()
+
+/mob/living/simple_animal/bot/hygienebot/proc/back_to_idle()
+	mode = BOT_IDLE
+	walk_to(src,0)
+	target = null
+	frustration = 0
+	last_found = world.time
+	stop_washing()
+	INVOKE_ASYNC(src, .proc/handle_automated_action)
+
+/mob/living/simple_animal/bot/hygienebot/proc/back_to_hunt()
+	frustration = 0
+	mode = BOT_HUNT
+	stop_washing()
+	INVOKE_ASYNC(src, .proc/handle_automated_action)
+
+/mob/living/simple_animal/bot/hygienebot/proc/look_for_lowhygiene()
+	for (var/mob/living/carbon/human/H in view(7,src)) //Find the NEET
+		if((H.name == oldtarget_name) && (world.time < last_found + 100))
+			continue
+		if(!check_purity(H)) //Theyre impure
+			target = H
+			oldtarget_name = H.name
+			speak("Unhygienic client found, Please stand still so I can clean you")
+			playsound(loc, 'sound/effects/hygienebot_happy.ogg', 60, 1)
+			visible_message("<b>[src]</b> points at [H.name]!")
+			mode = BOT_HUNT
+			INVOKE_ASYNC(src, .proc/handle_automated_action)
+			break
+		else
+			continue
+
+/mob/living/simple_animal/bot/hygienebot/proc/start_washing()
+	washing = TRUE
+	update_icon()
+
+/mob/living/simple_animal/bot/hygienebot/proc/stop_washing()
+	washing = FALSE
+	update_icon()
+
+
+
+/mob/living/simple_animal/bot/hygienebot/get_controls(mob/user)
+	var/dat
+	dat += hack(user)
+	dat += showpai(user)
+	dat += text({"
+<TT><B>Hygienebot X2 controls</B></TT><BR><BR>
+Status: []<BR>
+Behaviour controls are [locked ? "locked" : "unlocked"]<BR>
+Maintenance panel panel is [open ? "opened" : "closed"]"},
+
+"<A href='?src=[REF(src)];power=[TRUE]'>[on ? "On" : "Off"]</A>" )
+
+	if(!locked || issilicon(user) || IsAdminGhost(user))
+		dat += text({"<BR> Auto Patrol: []"},
+
+"<A href='?src=[REF(src)];operation=patrol'>[auto_patrol ? "On" : "Off"]</A>" )
+	return	dat
+
+/mob/living/simple_animal/bot/hygienebot/proc/check_purity(mob/living/L)
+	if(emagged && L.stat != DEAD)
+		return FALSE
+
+	var/obj/item/head = L.get_item_by_slot(ITEM_SLOT_HEAD)
```

Comment:
Please get rid of these duplicates with a loop

Reply:
how though

Reply:
for(var/slot in list(ITEM_SLOT_HEAD,ITEM_SLOT_MASK,...))